app/models/category.rb

```diff
@@ -664,6 +664,14 @@ def permissions_compatibility_validator
     end
   end
 
+  def self.ensure_consistency!
+    Category.all.each do |category|
```

Comment:
Even though we aren't supposed to have too much categories, it's always best to use "lazy" loading when iterating over an entire table.

```suggestion
    Category.find_each do |category|
```

Reply:
Like @SamSaffron said, you should also filter out categories which already have a topic using the `LEFT JOIN`.